### PR TITLE
[NFC][Clang] Address reviews about overrideFunctionFeaturesWithTargetFeatures

### PR DESCRIPTION
**Pull request by jmmartinez**
Addressing remarks after merge of D159257

* Add comment
* Remove irrelevant CHECKs from test
* Simplify function
* Use llvm::sort before setting target-features as it is done in CodeGenModeule

### CODE REVIEW

File: clang/test/CodeGen/link-builtin-bitcode.c [hunk 1]
```diff
@@ -43,7 +43,7 @@ int bar() { return no_attr() + attr_in_target() + attr_not_in_target() + attr_in
...
-// CHECK: attributes #[[ATTR_INCOMPATIBLE]] = { convergent noinline nounwind optnone "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="gfx90a" "target-features"="-gfx9-insts,+16-bit-insts,+atomic-buffer-global-pk-add-f16-insts,+atomic-fadd-rtn-insts,+ci-insts,+dl-insts,+dot1-insts,+dot10-insts,+dot2-insts,+dot3-insts,+dot4-insts,+dot5-insts,+dot6-insts,+dot7-insts,+dpp,+gfx8-insts,+gfx90a-insts,+gws,+image-insts,+mai-insts,+s-memrealtime,+s-memtime-inst,+wavefrontsize64" }
+// CHECK: attributes #[[ATTR_BAR]] = { {{.*}} }
+// CHECK: attributes #[[ATTR_COMPATIBLE]] = { {{.*}} }
+// CHECK: attributes #[[ATTR_EXTEND]] = { convergent noinline nounwind optnone "no-trapping-math"="true" "stack-protector-buffer-size"="8" "target-cpu"="gfx90a" "target-features"="+16-bit-insts,+atomic-buffer-global-pk-add-f16-insts,+atomic-fadd-rtn-insts,+ci-insts,+dl-insts,+dot1-insts,+dot10-insts,+dot2-insts,+dot3-insts,+dot4-insts,+dot5-insts,+dot6-insts,+dot7-insts,+dpp,+extended-image-insts,+gfx8-insts,+gfx9-insts,+gfx90a-insts,+gws,+image-insts,+mai-insts,+s-memrealtime,+s-memtime-inst,+wavefrontsize64" }
```
Comment (yxsamliu) [inline clang/test/CodeGen/link-builtin-bitcode.c:48]: I think we do not need to check attributes like 'convergent noinline nounwind optnone "no-trapping-math"="true" "stack-protector-buffer-size"="8"', etc. That's what I mean 'irrelevant attributes' in the Phbricator review.

But we still need to check "target-cpu" and "target-features" attributes, even for ATTR_BAR and ATTR_COMPATIBLE

Comment (jmmartinez) [reply clang/test/CodeGen/link-builtin-bitcode.c]: I've updated the test to match that behaviour.

**Comment by jmmartinez** [comment]
@yxsamliu Is it ok if I merge this patch?

**Review by yxsamliu — APPROVED** [review]
LGTM. Sorry for the delay

**Pull request merged**